### PR TITLE
[Improvement][Etcd] Support SSL In Etcd And Enhance Etcd In Helm

### LINKED ISSUE #13923: [Improvement][Etcd] Support SSL In Etcd And Enhance Etcd In Helm
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Support SSL In Etcd And Enhance Etcd In Helm

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by qingwli**
- close #13923

### CODE REVIEW

**Comment by zhongjiajie** [comment]
@kezhenxu94 do you have a time to take a look this PR?

File: dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-etcd/src/test/java/org/apache/dolphinscheduler/plugin/registry/etcd/EtcdRegistryTest.java [hunk 1]
```diff
@@ -140,4 +141,23 @@ public void notify(Event event) {
     public static void after() throws IOException {
         registry.close();
     }
+
+    @Disabled
+    @Test
```
Comment (kezhenxu94) [inline dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-etcd/src/test/java/org/apache/dolphinscheduler/plugin/registry/etcd/EtcdRegistryTest.java]: What's the point to add a `@Disabled` test?

Comment (qingwli) [reply dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-etcd/src/test/java/org/apache/dolphinscheduler/plugin/registry/etcd/EtcdRegistryTest.java]: This test method is used for the user to test connect etcd with configuration, user need to input their endpoints and SSL file.

File: deploy/kubernetes/dolphinscheduler/templates/secret-external-etcd-ssl.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+{{- if .Values.etcd.enabled }}
+apiVersion: v1
+kind: Secret
```
Comment (kezhenxu94) [inline deploy/kubernetes/dolphinscheduler/templates/secret-external-etcd-ssl.yaml]: etcd being enabled doesn't indicate ssl is enabled too, so the condition `{{- if .Values.etcd.enabled }}` to create this `Secret` doesn't seem to make sense

Comment (qingwli) [reply deploy/kubernetes/dolphinscheduler/templates/secret-external-etcd-ssl.yaml]: Ok, I will add a new var named ssl.enable

**Comment by qingwli** [comment]
PTAL @kezhenxu94

**Comment by qingwli** [comment]
PTAL @kezhenxu94

File: dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-etcd/src/test/java/org/apache/dolphinscheduler/plugin/registry/etcd/EtcdRegistryTest.java (hunk 1 above)
Comment (kezhenxu94) [reply dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-etcd/src/test/java/org/apache/dolphinscheduler/plugin/registry/etcd/EtcdRegistryTest.java]: > This test method is used for the user to test connect etcd with configuration, user need to input their endpoints and SSL file.

Etcd has its own method to test connection, users don't even have the source codes, this test should be removed, and the comments should go into the doc

Comment (qingwli) [reply dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-etcd/src/test/java/org/apache/dolphinscheduler/plugin/registry/etcd/EtcdRegistryTest.java]: The comments has exist in readme.md

**Review by kezhenxu94 — DISMISSED** [review]

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**